src/news_gui.cpp

```diff
@@ -999,7 +971,7 @@ static void RemoveOldNewsItems()
  */
 void ChangeVehicleNews(VehicleID from_index, VehicleID to_index)
 {
-	for (NewsItem *ni = _oldest_news; ni != nullptr; ni = ni->next) {
+	for (auto ni = std::begin(_news); ni != std::end(_news); ++ni) {
```

Comment:
Can't this be something like the following. Yeah, I know it's a bit of scope creep.
```suggestion
	for (auto ni : _news) {
```

Reply:
Good catch, range-for is preferred in my book.